torchtune/models/mistral/_component_builders.py

```diff
@@ -404,3 +404,212 @@ def lora_mistral_mlp(
         down_proj=down_proj,
         up_proj=up_proj,
     )
+
+
+def mistral_classifier(
```

Comment:
Another path we could consider here is providing an optional output dim argument in the existing Mistral builders and if it's not passed just set to vocab_size. I hope it's not too unintuitive, but could help get around your previous point regarding "classifier" now being a bit of a misnomer.

Reply:
I see what you're meaning. My only point would be that the implementation I have now minimises changes to the base mistral models. I think keeping it modular for now while we're still experimenting with how this model is going to be used downstream would be easiest, and we can try out potentially cleaner paths once things are more concrete.

Reply:
Yep this sounds good to me!